trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid_Module_README.md

```diff
@@ -0,0 +1,113 @@
+
+Before the final calculated release/outflow is returned from this module back to the model at each timestep, mass balance checks are
+performed to ensure the calculated release does not cause the reservoir storage to fall below the minimum or exceed the max, and the
+release is modified accordingly.
+
+
+### Input Parameters
+
+This module requires five input parameters that are set in hydro.namelist.
+
+* ```reservoir_persistence``` is a boolean parameter that will need to be set to ```.TRUE.``` for this module to be activated. This will cause the model to read the Q_Type variable
+from LAKEPARM.nc, the lake parameter file. The Q_Type for a hybrid reservoir is currently set to '2' in the lake parameter file.
+
+* ```reservoir_persistence_parameter_file``` is the NetCDF parameter file that holds the weights and corresponding gage ID for each lake ID.
+
+* ```reservoir_timeslice_path``` is the path to all the
+timeslice files used by this module.
+
+* ```reservoir_observation_lookback_hours``` is an integer parameter that specifies how many hours before the model start time the module will
+search for a corresponding timeslice file, and '18' would be a typical value and is the default.
+
+* ```reservoir_observation_update_time_interval_seconds``` is an integer parameter that determines how often the reservoirs will look for a new timeslice
+observation, and '86400', the number of seconds in a day, would be a typical value and is the default. If the model is set to run any
+type of forecast, then this parameter will automatically be set to 1,000,000,000
+seconds in order to prevent the module from reading any new timeslice observations after the first timestep.
+
+
+### Module Architecture
+
+This module builds off of the same class structure and architecture of the Level Pool module. As in Level Pool, each reservoir in this
+module is instantiated into an object at model initialization. The Persistence_Level_Pool_Hybrid directory contains
+module_persistence_levelpool_hybrid.F, module_persistence_levelpool_hybrid.F, module_persistence_levelpool_hybrid_parameters.F,
+module_persistence_levelpool_hybrid_state.F, module_persistence_levelpool_hybrid_tests.F, and a Makefile.
+
+
+The module_persistence_levelpool_hybrid.F module defines and instantiates objects for a hybrid persistence levelpool type
+reservoir. The module_RT.F module will call and pass parameters to the constructor in this module to instantiate the hybrid reservoir
+object and its sub-objects. The hybrid reservoir struct inherits input and output stucts from the reservoir base module and calls
+instantiation of these into sub-objects. The hybrid reservoir struct also points to structs for hybrid parameters and state and calls
+instantiation of these into sub-objects. A pointer to a levelpool reservoir object is also held in state, and this module
+instantiates that levelpool object. There is also a subroutine to run hybrid release that is derived from the reservoir base
```

Comment:
SO this module replaces level pool, so this levelpool object is not in addition to a normal level pool object also running. Might clarify that at runtime, if the persistence is selected, this module replaces regular level pool.. some where early on. It's sort of implied at the very top.

Reply:
Agreed. I will clarify.

Reply:
@jdmattern-noaa fixing